src/components/molecules/tables/eventTable/eventTable.tsx

```diff
@@ -5,15 +5,81 @@ import { useHistory } from 'react-router-dom';
 import Icon from 'components/atoms/icons/icon';
 import styles from './eventTable.module.scss';
 import { transformDate } from 'utils/timeConverter';
+import { useEffect, useState } from 'react';
+import Modal from 'components/molecules/modal/Modal';
+import ConformationBox from 'components/molecules/confirmationBox/ConfirmationBox';
+import useConfirmation from 'hooks/useConfirmation';
+import { deleteEvent } from 'api';
+import { useToast } from 'hooks/useToast';
 
 const EventTable = () => {
-  const { events } = useUpcomingEvents();
+  const { events, error, setFetch } = useUpcomingEvents();
+  const [deleteModal, setOpenDeleteModal] = useState(false);
+  const [selectedEvent, setSelected] = useState<
+    Pick<Event, 'eid' | 'title'> | undefined
+  >();
   const history = useHistory();
+  const { addToast } = useToast();
 
   const moveToEventAdminPage = (eid: string) => {
     history.push(`event/${eid}/admin`);
   };
 
+  const openDeleteColumn = (eid: string) => {
+    const selected = events?.find((mem) => {
+      return mem.eid === eid;
+    });
+    setSelected(selected);
+    setOpenDeleteModal(true);
+  };
+
+  const adminDeleteEvent = async () => {
+    setOpenDeleteModal(true);
+    try {
+      if (confirmed === false) {
+        setOpenDeleteModal(false);
+        return;
+      }
+      if (!selectedEvent) {
+        return;
+      }
+      await deleteEvent(selectedEvent.eid);
+      addToast({
+        title: 'Suksess',
+        status: 'success',
+        description: `${selectedEvent?.title} er slettet`,
+      });
+    } catch (error) {
+      // if another admin deletes the event
+      if (error.statusCode === 404) {
+        addToast({
+          title: 'Kunne ikke finne arrangement',
+          status: 'error',
+          description: `kunne ikke finne ${selectedEvent?.title}. Prøv å oppdater siden`,
+        });
+      }
+      addToast({
+        title: 'En uforutsett feil skjedde',
+        status: 'error',
+        description: 'Problemer med slettingen av arrangementet',
+      });
+    }
+    setFetch(true);
+    setOpenDeleteModal(false);
+  };
+
+  useEffect(() => {
+    // display error toast on error caught by useUpcomingEvents hook
+    if (error !== null) {
+      addToast({
+        title: 'Kunne ikke laste inn arrangementetene',
+        status: 'error',
+      });
+    }
+  }, [error]);
+
+  const { confirmed, setConfirmed } = useConfirmation(adminDeleteEvent);
```

Comment:
Why do we need this? Can't we just call `adminDeleteEvent` in the `onAccept` prop in the confirmation component?

`onDecline` could be set to an optional prop in the component so we don't do anything in that case

Then we can also remove the check for confirmation in the delete function

Reply:
Yes that would be smoother, changing now